src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java

```diff
@@ -130,32 +130,39 @@ default ThrowableProblem toProblem(final Throwable throwable, final StatusType s
     }
 
     default ThrowableProblem toProblem(final Throwable throwable, final StatusType status, final URI type) {
-        final Throwable cause = throwable.getCause();
+        final ThrowableProblem problem = prepare(throwable, type, status).build();
+        final StackTraceElement[] stackTrace = createStackTrace(throwable);
+        problem.setStackTrace(stackTrace);
+        return problem;
+    }
 
-        final ProblemBuilder builder = Problem.builder()
+    default ProblemBuilder prepare(final Throwable throwable, final URI type, final StatusType status) {
```

Comment:
I would make arg order the same as in `toProblem`

Reply:
Done